azkaban-hadoop-security-plugin/build.gradle

```diff
@@ -1,12 +1,20 @@
 apply plugin: 'distribution'
 
+configurations {
+    testCompile.extendsFrom compileOnly
```

Comment:
> Modifying the sourceSets is not an issue and is expected.

AFAIK, I didn't modify any source set. I don't understand the context here.

> I thought extendsFrom would replace the existing superConfigs. After some experiments, I realized that it adds to the existing configs.

Yes. It simply _extends_ the configuration/collection with another.

> Also since the build only packages the jar and not the run time dependencies, does it need to use compileOnly in the first place? What's your long term plan?

Correction: build packages the jar WITH the runtime dependencies. `compileOnly` means that this collection is only used for compiling the artifact and they are not added to runtime. It does not need to be `compileOnly` since HDFS storage introduced Hadoop as a compile level dependency at this time. But prior to that, Hadoop and Hive dependencies were not shipped with Azkaban.

The long term plan IMO should be to get rid of Hadoop, Hive etc dependencies from core Azkaban. HDFS Storage and Hadoop Security Manager code should run in a plugin module like environment.